agent/eventhandler/task_handler.go

```diff
@@ -77,9 +79,14 @@ type TaskHandler struct {
 	minDrainEventsFrequency time.Duration
 	maxDrainEventsFrequency time.Duration
 
-	state  dockerstate.TaskEngineState
-	client api.ECSClient
-	ctx    context.Context
+	state              dockerstate.TaskEngineState
+	client             api.ECSClient
+	ctx                context.Context
+	cfg                *config.Config
+	delay              time.Duration
```

Comment:
like the comment above, `delay` is a bit generic of a name since this is specifically only used for disconnected mode task event flow retry backoff (IIUC)